### ISSUE TITLE
It is possible to store either structs or pointer to structs ?

### ISSUE THREAD

**Issue by estebanzimanyi** [rurban/ctl#25]
In STL it is possible to define, e.g.,
```
vector<SweepEvent> ev;
vector<SweepEvent*> evp;
```
After looking in the documentation, the code, and the examples I was not able to determine whether this can be done in CTL.
Thanks for your answer.
Esteban

**Comment by Junming-Liang** [comment]
I, too, have entertained similar contemplations. You can encapsulate pointers within the struct to achieve the sought-after effect.
```
typedef struct SweepEventPoint{ SweepEvent *point};
 #define POD
 #define NOT_INTEGRAL
 #define T SweepEventPoint
 #include "ctl/vector.h"
```

**Comment by rurban** [comment]
Yes. CTL instantiates a container from a named `T` token, so structs work directly and pointer element types need a typedef.

```c
#include <string.h>

typedef struct SweepEvent {
    int id;
} SweepEvent;
typedef SweepEvent *SweepEventPtr;

#define POD
#define NOT_INTEGRAL
#define T SweepEvent
#include <ctl/vector.h>

#define POD
#define NOT_INTEGRAL
#define T SweepEventPtr
#include <ctl/vector.h>

vec_SweepEvent ev = vec_SweepEvent_init();
vec_SweepEventPtr evp = vec_SweepEventPtr_init();
```

Do not use `#define T SweepEvent *`: `T` is also pasted into generated identifiers such as `vec_T`, so it must be a single identifier. `SweepEventPtr` gives the pointer specialization the generated name `vec_SweepEventPtr`.

`POD` means CTL performs bitwise copies and does not own the pointed-to `SweepEvent` objects; callers retain responsibility for their lifetime. `NOT_INTEGRAL` is required for a struct type because CTL must not generate integer comparisons and hashing for it. If `SweepEvent` owns resources, omit `POD` and provide the usual `SweepEvent_copy` / `SweepEvent_free` hooks instead. Set `compare` or `equal` when using operations that need value comparison.

**Comment by rurban** [comment]
Added documentation and a regression test for named struct and pointer element types. The answer above describes the required typedef and ownership contract. Closing.
